### PR TITLE
Switch to setuptools entrypoint console scripts for filters

### PR DESCRIPTION
**Pull request by mtreinish**
This commit switches from using scripts to setuptools entrypoint console
scripts for the filter scripts. This should fix the issues for windows
users trying to execute the scripts. To accomplish this the filter
scripts are moved into the subunit package namespace and a callable to
run the script is added to each module.

Closes-Bug: #1322888

### CODE REVIEW

File: python/subunit/filter_scripts/subunit2disk.py [hunk 1]
```diff
@@ -19,5 +19,9 @@
 from subunit._to_disk import to_disk
 
 
-if __name__ == '__main__':
+def main():
     exit(to_disk())
```
Comment (42B) [inline python/subunit/filter_scripts/subunit2disk.py]: Shouldn't this be `sys.exit(...)`?

File: python/subunit/filter_scripts/subunit2gtk.py [hunk 2]
```diff
@@ -218,23 +218,28 @@ def update_counts(self):
...
+    gtk.main()
+    if result.decorated.wasSuccessful():
+        exit_code = 0
+    else:
+        exit_code = 1
+    sys.exit(exit_code)
+
+
+if __name__ == '__main__':
+    main(
```
Comment (42B) [inline python/subunit/filter_scripts/subunit2gtk.py]: Missing closing bracket for `main()`?

File: python/subunit/filter_scripts/subunit2disk.py (hunk 1 above)
Comment (42B) [reply python/subunit/filter_scripts/subunit2disk.py]: > The site module (which is imported automatically during startup, except if the -S command-line option is given) adds several constants to the built-in namespace. ***They are useful for the interactive interpreter shell and should not be used in programs.***

* https://docs.python.org/3/library/constants.html?highlight=exit#constants-added-by-the-site-module

**Comment by mtreinish** [comment]
@rbtcollins any pointers on how to combine this with the Makefile? (which is why travis is failing) I took a quick look at the Makefile.am and it's defining the filter scripts at the old location. I can delete those if that is what's needed. But it's not clear to me how it interacts with the setup.py and I feel like there is a larger change needed to have this work properly with the Makefile.

**Comment by jelmer** [comment]
This change in general SGTM; it probably needs a rebase though, and travis is unhappy.

**Pull request merged**